privacyidea/lib/eventhandler/tokenhandler.py

```diff
@@ -628,14 +628,16 @@ def do(self, action, options=None):
                           "genkey": 1,
                           "realm": handler_options.get("realm", "")}
             user = None
+
+            # Some tokentypes need additional parameters
+            if handler_options.get("additional_params"):
+                add_params = yaml.safe_load(handler_options.get("additional_params"))
+                if isinstance(add_params, dict):
```

Comment:
We should probably wrap the whole `if action == INIT` block in a try-except block. Then we can be sure that the result of `yaml.safe_load` is a `dict` or `None`.
Also the info-message in the else-block in line 621 is misleading. The check for serials should be inside the if-block before.